File: androidApp/src/main/java/de/stefan/lang/shapebyte/android/shared/buttons/ui/RoundedImageButtonAppearance.kt
package de.stefan.lang.shapebyte.android.shared.buttons.ui

import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import de.stefan.lang.designsystem.theme.ThemeAdditions

enum class RoundedImageButtonAppearance(val size: Dp) {
    Small((ThemeAdditions.dimensions.medium - ThemeAdditions.dimensions.xTiny).dp),
    Medium(ThemeAdditions.dimensions.medium.dp),
    Large(ThemeAdditions.dimensions.xLarge.dp),
}

File: iosApp/iosApp/shared/buttons/RoundedImageButtonAppearance.swift
import SwiftUI

enum RoundedImageButtonAppearance: CaseIterable {
    case small
    case medium
    case large

    var size: Double {
        let retVal: Double
        switch self {
        case .small:
            retVal = .dimensionMedium - .dimensionXTiny
        case .medium:
                retVal = .dimensionMedium
        case .large:
                retVal = .dimensionXLarge
        }

        return retVal
    }
}
